source/docs/example-projects/halloween-treasure-hunt.rst

```diff
@@ -0,0 +1,251 @@
+every time we click the *pumpkin*. While players will totally love this, we should probably avoid giving out unlimited 
+diamonds. That's easy to fix though, we just have to *remove* the pumpkin once it's clicked. If we do it before even 
+giving out the reward, we'll make sure it won't be clicked twice. 
+
+For this, we'll use the :guilabel:`modifyblock` command, which lets us specify a *location* and a *material*. Now we 
+only need to know which location was clicked by the player. Time to make use of *context* tags! This kind of tags are 
+event specific and will let us retrieve useful information from said event. If we check again the event's 
+documentation, we can see it has a ``<context.location>`` tag available, which is just what we needed for the first 
+argument. The material, on the other hand, will be just ``air`` as we want to remove the original pumpkin. The full 
+command line will then be ``- modifyblock <context.location> air``.
+
+Our script with these new commands should look like this:
+
+.. code-block:: dscript
+    :name: figure_1_3
+    :linenos:
+    :emphasize-lines: 7,8
+
+    Halloween_Treasure_Hunt:
+    type: world
+    debug: true
+    events:
+      on player left clicks pumpkin in Hub:
+      - narrate "yay"
+      - modifyblock <context.location> air
+      - give diamond
+
+.. rst-class:: figurecaption
+
+**Figure 1.3** Our world script with core functionality
+
+Rinse and repeat: save, reload scripts and do a quick test. Amazing! This deserves a "yay". Speaking of yays… we don't 
+need to narrate ``yay`` for testing purposes anymore, so we better change it to something more informative. Something 
+like ``- narrate "You've found a pumpkin! Here's your reward!"`` sounds like the way to go.
+
+Topping It Off
+==============
+
+Let's make it even more fun. What if *jack o' lanterns* gave a diamond to *every online player*? Yeah, we can make 
+that happen too! Let's start by making a copy of the event we already have and its contents. We should now change the 
+``pumpkin`` material of said event to ``jack_o_lantern``, so it's only triggered when clicking jack o' lantern blocks.
+
+.. note::
+    There are other ways to achieve the same result. For example, a single general event that is triggered for every 
+    block clicked could be used. This would mean filtering the needed blocks with logic afterwards, usually with 
+    **if/else if/else** trees or **choose** commands. In this guide though, two separate events will be used as that 
+    can help keep it simple without functionality.
+
+Inside the event, we need to repeat the give command once per player. How to do that? You've guessed it, a loop! In 
+our case, to wrap the :guilabel:`give` command with a :guilabel:`foreach` loop is all we need. This loop takes a 
+*list* when it starts and executes some commands for *every object* on the list. We just need to feed it the list of 
+online players, which can be accessed through ``<server.list_online_players>``.
```

Comment:
You can just briefly note that the `<server.list_online_players>` tag returns the list of all online players. The details on how the `foreach` command works will be covered in a separate section. (Optionally, add a TODO placeholder here noting that the foreach command section needs to be done.)